tests/README.md

```diff
@@ -0,0 +1,323 @@
+<!-- Integration in CONP-Portal -->
+
+## Authenticated Dataset
+
+<!-- What should be done prior the tests -->
+
+The test suite currently supports restricted dataset with authentication through Loris or Zenodo, however, it requires prior setup described below. For completing the setup, please contact the CircleCI administrator: Tristan Glatard.
+
+### Secrets creation
+
+To create the secrets in CircleCI you will need to generate a standardized \${PROJECT_NAME} as done below:
+
+```python
+from tests.functions import project_name2env
+project_name2env("projects/datatset_name".split("/")[-1])
+# DATATSET_NAME
+```
+
+#### Loris
+
+1. Create a Loris account to be used on CircleCI.
+1. Assisted by the CircleCI administrator create the following secrets:
+   - `${PROJECT_NAME}_USERNAME`
+   - `${PROJECT_NAME}_PASSWORD`
+   - `${PROJECT_NAME}_LORIS_API`; e.g. `https://example.loris.ca/api/v0.0.0`
+
+#### Zenodo
+
+1.  [Create a new Zenodo token](https://zenodo.org/account/settings/applications/tokens/new/)
+1.  Assisted by the CircleCI administrator, create a secret called `${PROJECT_NAME}_ZENODO_TOKEN`.
+
+### Limitations of secrets
+
+A major limitation of secrets is that authenticated datasets cannot be tested on pull requests due to privacy issues.
+Indeed, malicious users could easily retrieve secrets by making a pull requests.
+
+To avoid this problematic, the user making a pull request on an authenticated dataset can set up secrets for the dataset in its CircleCI account.
+
+#TODO
+Otherwise, the pull requests for authenticated datasets should be merged into a devel branch to assure their proper functioning.
+Then, when everything is working, it can be merged into the master branch.
+
+## Implementation Keypoints
+
+### Flaky test
+
+The current test suite cover a broad range of components for dataset to work properly. Unfortunately, there are components that are hard to test in real environment such as downloading files from live servers.
+<br/>
+For this reason, we opted to allow test to be flaky.
+That is, when a test fails it will rerun up to 3 times.
+There is a 5 seconds delay between rerun of failures.
+Furthermore, to avoid Datalad command to conflict during a dataset installation, an instruction to install a dataset will be queued if this dataset is already installing.
+
+### Empty dataset
+
+When a dataset has no files contained in its annex, the test suite assume the dataset works properly and pass the test.
+
+### Timeout
+
+Since the test suite involves downloading data from external servers, the runtime of the test suite can vary considerably when there is network issues.
+To mitigate bad connection to server, the test suite imposes a 10-minute timeout for each test as well as a 2-minute timeout specific to each dataset download.
+
+### Size of annexed file
+
+Dataset files vary in size.
+This can be and issue when testing the download of a dataset as downloading a large file might result in timeout and test failure.
```

Comment:
```suggestion
This can be an issue when testing the download of a dataset as downloading a large file might result in a timeout and test failure.
```